README.md

```diff
@@ -6,6 +6,42 @@ hperf is a tool for active measurements of the maximum achievable bandwidth betw
 Hperf was made to test networks in large infrastructure. It's highly scalable and cabaple of running parallel tests over
 a long period of time. 
 
+## Common use cases
+- Debugging link/nic MTU issues
+- Optimizing throughput speed for specific payload/buffer sizes
+- Finding servers that present latency on the application level when ping is showing no latency
+- Testing overall network throughput
+- Testing server to server connectivity 
+
+## Core concepts
+### The hperf binary
+The binary can act as both client and server.
+
+### Client
+The client part of hperf is responsible for orchestrating the servers. Its only job is to send commands to the
+servers and receive incremental stats updates. It can be executed from any machine that can talk to the servers.
+
+### Servers
+Servers are the machines we are testing. To launch the hperf command in servers mode, simply use the `server` command:
+NOTE: `server` is the only command you can execute on the servers. All other commands are executed from the client.
+```bash
+$ ./hperf server --help
+```
+This command will start an API and websocket on the given `--address` and save test results to `--storage-path`. 
+
+### The listen command
+Hperf has been modified to work very differently than how it was before. The new version can run tests without a
```

Comment:
Maybe just focus on how it is now... I don't know how it was before (and frankly don't care) Maybe add a separate section on it, if needed.

Also nothing here really explains what `listen` actually does, AFAICT.

Reply:
No we should just document what is latest, no one knows what hperf was before.